client/src/components/SaveRestaurant.jsx

```diff
@@ -4,16 +4,41 @@ class SaveRestaurant extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-
+      saveRestaurant: false,
+    }
+    this.clickSave = this.clickSave.bind(this);
+  }
+  clickSave() {
+    if (this.state.saveRestaurant) {
```

Comment:
Hey I think you should know about the ! method for toggling--it will get rid of the need for the if/else conditional. You can just make the current state equal to the opposite of the previous state.